core/src/banking_stage.rs

```diff
@@ -1982,18 +1979,27 @@ impl BankingStage {
     fn receive_until(
         verified_receiver: &CrossbeamReceiver<Vec<PacketBatch>>,
         recv_timeout: Duration,
-        batching_timeout: Duration,
-        batch_size_upperbound: usize,
+        packet_count_upperbound: usize,
     ) -> Result<Vec<PacketBatch>, RecvTimeoutError> {
         let start = Instant::now();
         let mut packet_batches = verified_receiver.recv_timeout(recv_timeout)?;
+        let mut num_packets_received: usize =
```

Comment:
Can we just do 1 sum at the end?

Reply:
It's calculating how many packets are received across all batches. So the count has to be done for each packet batch, and then accumulated. What do you mean by one sum at the end?

Reply:
Also, the cumulative count is being used to figure out when to break from the receive loop. So it has to be updated on every receive.

Reply:
Do we need to bound the number of packet batches, or do we just want to bound the amount of time we spend listening for packets? Can we try just with a time budget but no packet budget?

Reply:
Having a packet bound was suggested in one of the discord discussions. I think it makes sense to have a back stop there, as in Mainnet they have seen issues when banking stage received too many packets in one iteration.

What is the concern with having a check for packet count? In case we receive more packets than we can buffer, those will be dropped anyway. This check prevents that from happening.

Reply:
I was mostly just wondering what the motivation was. Totally makes sense though that we'd want to alleviate some downstream pressure. Generally LGTM.